src/tests/ftest/util/test_utils_pool.py

```diff
@@ -10,21 +10,100 @@
 from command_utils import BasicParameter
 from exception_utils import CommandFailure
 from pydaos.raw import (DaosApiError, DaosPool, c_uuid_to_str, daos_cref)
 from general_utils import check_pool_files, DaosTestError
 from server_utils_base import ServerFailed, AutosizeCancel
 from dmg_utils import DmgCommand
 
+POOL_NAMESPACE = "/run/pool/*"
+
+
+def add_pool(test, namespace=POOL_NAMESPACE, create=True, connect=True, index=0, **params):
+    """Add a new TestPool object to the test.
+
+    Args:
+        test (Test): the test to which the pool will be added
+        namespace (str, optional): TestPool parameters path in the test yaml file. Defaults to
+            POOL_NAMESPACE.
+        create (bool, optional): should the pool be created. Defaults to True.
+        connect (bool, optional): should the pool be connected. Defaults to True.
+        index (int, optional): Server index for dmg command. Defaults to 0.
+
+    Returns:
+        TestPool: the new pool object
+
+    """
+    pool = TestPool(
+        namespace=namespace, context=test.context, dmg_command=test.get_dmg_command(index),
+        label_generator=test.label_generator)
+    pool.get_params(test)
+    if params:
+        pool.update_params(**params)
+    if create:
+        pool.create()
+    if create and connect:
+        pool.connect()
+
+    # Add a step to remove this pool when the test completes and ensure their is enough time for the
+    # pool destroy to be attempted - accounting for a possible dmg command timeout
+    test.increment_timeout(200)
+    test.register_cleanup(**pool.get_cleanup_entry(test))
+
+    return pool
+
+
+def remove_pool(test, pool):
+    """Remove the requested pool from the test.
+
+    Args:
+        test (Test): the test from which to destroy the pool
+        pool (TestPool): the pool to destroy
+
+    Returns:
+        list: a list of any errors detected when removing the pool
+
+    """
+    error_list = []
+    test.test_log.info("Destroying pool %s", pool.identifier)
+
+    # Ensure exceptions are raised for any failed command
+    exit_status_exception = None
+    if pool.dmg is not None:
+        exit_status_exception = pool.dmg.exit_status_exception
+        pool.dmg.exit_status_exception = True
+
+    # Only disconnect a pool that has been connected by the test
+    if not hasattr(pool, "connected") or pool.connected:
+        try:
+            pool.disconnect()
+        except (DaosApiError, TestFail) as error:
+            test.test_log.info("  {}".format(error))
+            error_list.append("Error disconnecting pool {}: {}".format(pool.identifier, error))
+
+    # Only destroy a pool that has been created by the test
+    if not hasattr(pool, "attached") or pool.attached:
+        try:
+            pool.destroy(1)
+        except (DaosApiError, TestFail) as error:
+            test.test_log.info("  {}".format(error))
+            error_list.append("Error destroying pool {}: {}".format(pool.identifier, error))
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string